conan/tools/microsoft/toolchain.py

```diff
@@ -27,6 +28,21 @@ def _name_condition(settings):
         return name.lower(), condition
 
     def write_toolchain_files(self):
+        # Warning
+        msg = ("\n*****************************************************************\n"
+               "******************************************************************\n"
+               "'write_toolchain_files()' has been deprecated and moved.\n"
+               "It will be removed in next Conan release.\n"
+               "Use 'generate()' method instead.\n"
+               "********************************************************************\n"
+               "********************************************************************\n")
+        from conans.client.output import Color, ConanOutput
+        ConanOutput(self._conanfile.output._stream,
+                    color=self._conanfile.output._color).writeln(msg, front=Color.BRIGHT_RED)
+        warnings.warn(msg)
```

Comment:
We might add the `category=DeprecationWarning` here, but probably it doesn't worth it (it requires changes in the test to capture it)

Reply:
Yes, if it is going to be removed soon, probably not worth the effort of changing tests forth and back.